service/resource/cloudformation/resource.go

```diff
@@ -110,3 +110,13 @@ func toDeleteStackInput(v interface{}) (awscloudformation.DeleteStackInput, erro
 
 	return deleteStackInput, nil
 }
+
+func getStackOutputValue(outputs []*awscloudformation.Output, key string) (string, error) {
+	for _, o := range outputs {
+		if *o.OutputKey == key {
+			return *o.OutputValue, nil
+		}
+	}
+
+	return "", notFoundError
```

Comment:
Please always mask errors.

Reply:
Sure, i'm doing it after receiving the error https://github.com/giantswarm/aws-operator/pull/502/files#diff-4313bcb330a12c7c826a88a7090a0905R46 what is the criteria here, always mask at the lower level and let it bubble up untouched? or remask at each stage?

Reply:
Masking everywhere.

Reply:
Ok done, could you please elaborate on the reason for masking more than once?

Reply:
Imagine your code handles errors in three places. When you do not mask you get an error message. How do you find the problem and the code that produced the error? Similar problem when masking only once. When you mask at the end you have the same problem than not masking at all. When masking somewhere in the middle of the stack you know where the error was within one place of the call stack but you do not know where it originates from nor where it was handled finally. When you mask once where the error can be produced you also miss the rest of the call stack which is often required to debug and solve an issue or at least to understand the situation.

Reply:
Ok, I agree with the reasons for not masking in places different where the error originates, but masking it at creation the full stack of how you got there should be available AIUI, am i missing something?

Anyway this question is not related to this PR, should we go ahead with it?

Reply:
I am happy to have better ways of doing it. Feel free to provide alternatives. Until then we should keep the convention up as it is.